webserver/webserver/src/main/java/io/helidon/webserver/RequestPredicate.java

```diff
@@ -16,226 +16,508 @@
+                .all(name)
+                .stream()
+                .anyMatch(value::equals));
+    }
 
     /**
-     * Accepts only requests of any specified content types.
+     * Accept request only when the specified cookie exists.
      *
-     * @param contentType Content type.
-     * @return New enhanced instance.
+     * @param name cookie name
+     * @return composed predicate representing the logical expression of the previous
+     * predicate expression <b>and</b> the logical expression implemented by this
+     * method
+     * @throws NullPointerException if the specified name is null
      */
-    RequestPredicate hasContentType(String... contentType);
+    public RequestPredicate containsCookie(final String name) {
+        Objects.requireNonNull(name, "cookie name");
+        return and((req) -> req.headers()
+                .values("cookie")
+                .stream()
+                .anyMatch((c) -> c.startsWith(name + "=")));
+    }
 
     /**
-     * Accepts by free form condition. Equivalent method for {@link #and(Predicate)}.
+     * Accept requests only when the specified cookie is valid.
      *
-     * @param requestPredicate A request predicate.
-     * @return New enhanced instance.
+     * @param name cookie name
+     * @param predicate predicate to match the cookie value
+     * @return new predicate representing the logical expression of the previous
+     * predicate expression <b>and</b> the logical expression implemented by this
+     * method
+     * @throws NullPointerException if the specified name or predicate is null
      */
-    RequestPredicate is(Predicate<? super ServerRequest> requestPredicate);
+    public RequestPredicate containsCookie(final String name,
+            final Predicate<String> predicate) {
+
+        Objects.requireNonNull(name, "cookie name");
+        Objects.requireNonNull(predicate, "cookie predicate");
+        return and((req) -> req.headers()
+                .cookies()
+                .all(name)
+                .stream()
+                .anyMatch((c) -> predicate.test(c)));
+    }
 
     /**
-     * Creates request-response handler/filter which calls provided handler only if this predicate accepts provided request,
-     * otherwise call {@link ServerRequest#next()} method.
+     * Accept requests only when the specified cookie contains a given value.
      *
-     * @param handler to apply if this instance accepts provided request.
-     * @return a conditional handler.
+     * @param name cookie name
+     * @param value expected cookie value
+     * @return composed predicate representing the logical expression of the previous
+     * predicate expression <b>and</b> the logical expression implemented by this
+     * method
+     * @throws NullPointerException if the specified name or value is null
      */
-    ConditionalHandler thenApply(Handler handler);
+    public RequestPredicate containsCookie(final String name,
+            final String value) {
```

Comment:
Similar thought here -- can this method just delegate to the other one?

Reply:
Yes, will make the change.